projects/components/src/datagrid/interfaces/datagrid-column.interface.ts

```diff
@@ -40,6 +40,106 @@ export enum GridColumnSortDirection {
  */
 export type FunctionRenderer<T> = (record: T) => string;
 
+/**
+ * A type of button who's displayability does not depend on the selected entity.
+ */
+export interface GlobalButton {
+    /**
+     * The translated text of the button.
+     */
+    label: string;
+    /**
+     * The function that is called when the button is pressed.
+     */
+    handler: () => void;
+    /**
+     * The function that is called to determine if the button should be displayed.
+     */
+    shouldDisplay: () => boolean;
+    /**
+     * The css class the button should have.
+     */
+    class?: string;
+}
+
+/**
+ * A type of button who's displayability dependends on the selected entity.
+ */
+export interface ContextualButton<R> {
+    /**
+     * The translated text of the button.
+     */
+    label: string;
+    /**
+     * The function that is called when the button is pressed.
+     *
+     * @param entity the currently selected entities.
+     */
+    handler: (entity: R[]) => void;
+    /**
+     * The function that is called to determine if the button should be displayed.
+     *
+     * @param entity the currently selected entities.
+     */
+    shouldDisplay: (rec: R[]) => boolean;
+    /**
+     * The css class the button should have.
+     */
+    class?: string;
+    /**
+     * The unique ID of this button.
+     */
+    id: string;
+    /**
+     * The Clarity icon of the contextual button that is displayed if the button is featured.
+     */
+    icon: string;
+}
+
+/**
+ * An enum that describes where the contextual buttons should display.
+ */
+export enum ContextualButtonPosition {
+    TOP = 'TOP',
+    ROW = 'ROW',
+}
+
+/**
+ * A configuration that descibes all the information about the contextual buttons.
+ */
+export interface ContextualButtonConfig<R> {
+    /**
+     * A list of all the contextual buttons.
+     */
+    buttons: ContextualButton<R>[];
+    /**
+     * An ordered list of {@link ContextualButton.id}s of buttons that should be in a featured position.
+     */
+    featured: string[];
+    /**
+     * How many buttons should display on the featured section.
```

Comment:
How is that different from the length of featured list above? Can you please add that to the comment?

Reply:
Tried to clarify this, let me know what you think

Reply:
Sorry, I don't understand the comment. If I remember right... you said this property is to select first 'n' non-disabled buttons from the featured list. And you said that's necessary to save the screen's real state.

Reply:
I am sorry I don't understand the comment. As I remember, you said that it is for saving horizontal real estate on the screen by choosing that many number of non-disabled featured buttons from the featured list. Right?